src/cursor/cur_version.c

```diff
@@ -308,37 +308,35 @@ __curversion_next_int(WT_CURSOR *cursor)
         }
 
         /* Get the ondisk value. */
-        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          NULL
-#endif
-        );
-        WT_ERR(ret);
-
-        if (!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
-            durable_stop_ts = version_cursor->upd_durable_stop_ts;
-            stop_ts = version_cursor->upd_stop_ts;
-            stop_txn = version_cursor->upd_stop_txnid;
-        } else {
-            durable_stop_ts = cbt->upd_value->tw.durable_stop_ts;
-            stop_ts = cbt->upd_value->tw.stop_ts;
-            stop_txn = cbt->upd_value->tw.stop_txn;
-        }
+        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw);
+        if (ret == 0) {
+            if (!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
+                durable_stop_ts = version_cursor->upd_durable_stop_ts;
+                stop_ts = version_cursor->upd_stop_ts;
+                stop_txn = version_cursor->upd_stop_txnid;
+            } else {
+                durable_stop_ts = cbt->upd_value->tw.durable_stop_ts;
+                stop_ts = cbt->upd_value->tw.stop_ts;
+                stop_txn = cbt->upd_value->tw.stop_txn;
+            }
 
-        if (tombstone != NULL &&
-          (tombstone->prepare_state == WT_PREPARE_INPROGRESS ||
-            tombstone->prepare_state == WT_PREPARE_LOCKED))
-            version_prepare_state = 1;
-        else
-            version_prepare_state = cbt->upd_value->tw.prepare;
+            if (tombstone != NULL &&
+              (tombstone->prepare_state == WT_PREPARE_INPROGRESS ||
+                tombstone->prepare_state == WT_PREPARE_LOCKED))
+                version_prepare_state = 1;
+            else
+                version_prepare_state = cbt->upd_value->tw.prepare;
 
-        WT_ERR(__curversion_set_value_with_format(cursor, WT_CURVERSION_METADATA_FORMAT,
-          cbt->upd_value->tw.start_txn, cbt->upd_value->tw.start_ts,
-          cbt->upd_value->tw.durable_start_ts, stop_txn, stop_ts, durable_stop_ts,
-          WT_UPDATE_STANDARD, version_prepare_state, 0, WT_CURVERSION_DISK_IMAGE));
+            WT_ERR(__curversion_set_value_with_format(cursor, WT_CURVERSION_METADATA_FORMAT,
+              cbt->upd_value->tw.start_txn, cbt->upd_value->tw.start_ts,
+              cbt->upd_value->tw.durable_start_ts, stop_txn, stop_ts, durable_stop_ts,
+              WT_UPDATE_STANDARD, version_prepare_state, 0, WT_CURVERSION_DISK_IMAGE));
+
+            upd_found = true;
+        } else if (ret != WT_RESTART)
```

Comment:
If the result is `WT_RESTART`, do we want to exit the function or continue its executing?

Reply:
We want to continue to look at the versions in the history store.

Reply:
Reading the code below... We might overwrite the `ret` value to `WT_NOTFOUND`:
```
    if (!upd_found)
        ret = WT_NOTFOUND;
```

Reply:
We should not return WT_RESTART to the user. I think we should set the ret to 0 if we see a restart error to explicit ignore it.

Reply:
Are you thinking of doing this?
```
    if (!upd_found)
        ret = WT_NOTFOUND;
    else {
        cbt->upd_value->type = WT_UPDATE_STANDARD;
        __wt_value_return(cbt, cbt->upd_value);
       if(ret == WT_RESTART) ret = 0; <------
    }
```

Reply:
```
    upd_found = true;
        } else if (ret == WT_RESTART)
            /* We may see an overflow removed cell. Ignore it. */
            ret = 0;
        else
            goto err;
```
No, return not found is ok. I just think code is better if we explicitly set the ret value to 0 if we see restart to ignore it. There is no beheavior change.